snaps/build-and-release-k8s-snaps.sh

```diff
@@ -17,7 +18,13 @@ git clone https://github.com/juju-solutions/release.git --branch rye/snaps --dep
 )
 
 rm -rf ./cdk-addons
-git clone https://github.com/juju-solutions/cdk-addons.git --depth 1
+if git ls-remote --exit-code --heads https://github.com/juju-solutions/cdk-addons.git ${ADDONS_BRANCH_VERSION}
+then
+  git clone https://github.com/juju-solutions/cdk-addons.git --branch ${ADDONS_BRANCH_VERSION} --depth 1
```

Comment:
Do we want to add an echo here as well to say that we found the branch and we are using it?